### PR TITLE
 Update containers, nf trace compatible now. Some to biocontainers.

### PR DESCRIPTION
**Pull request by mberacochea**
This branch has a the following:
- Moved over to quay.io (dockerhub has pull limits now)
- Migrated some tools to biocontainers, less maintaince work
- Added the tools required for Nextflow monitoring and trace

TODO: complete the build.py command to push all the containers.
I would like to use gh actions to push the containers when
we get a new release.

NOTE: it was built on top of https://github.com/EBI-Metagenomics/emg-viral-pipeline/pull/80

Should close https://github.com/EBI-Metagenomics/emg-viral-pipeline/issues/50

### CODE REVIEW

File: containers/README.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
-# Docker files for EMG Viral Pipeline
+# Containers for VIRify
+
+## Container registry
+
+The containers used on this pipeline are stored under the [Microbiome Informatics]() (Quay organization)[https://quay.io/repository/microbiome-informatics/].
+
+## Build process
+
+The build process takes care of tagging the images using the LABELS on the Dockerfiles of the tools.
```
Comment (hoelzer) [inline containers/README.md:9]: yeah, as said in the other PR you could use the LABEL + md5 sum of the Dockerfile for tagging. Then you can have multiple containers with the same tool version but also with other changes (added dependencies, ...)

**Review by hoelzer — APPROVED** [review]
Nice! +1

Ad automation container build

> TODO: complete the build.py command to push all the containers. I would like to use gh actions to push the containers when we get a new release.

That would be great. Would you like to set it up so that all the Dockerfiles are checked, and when there is a VERSION change (or different checksum), the container(s) get auto-build and pushed? This should work via gh Action somehow. But so far I only did this via having a single repository per Dockerfile. And then, when this Dockerfile has been updated the gh Action builds a new container and does the push.

**Comment by mberacochea** [comment]
This one also closes #74

File: containers/README.md (hunk 1 above)
Comment (mberacochea) [reply containers/README.md:9]: Yeah... that is a good idea -> https://github.com/EBI-Metagenomics/emg-viral-pipeline/issues/85

**Pull request merged**